scripts/bert/finetune_classifier.py

```diff
@@ -243,145 +248,151 @@
 do_lower_case = 'uncased' in dataset
 bert_tokenizer = BERTTokenizer(vocabulary, lower=do_lower_case)
 
-
 def preprocess_data(tokenizer, task, batch_size, dev_batch_size, max_len, pad=False):
-    """Data preparation function."""
-    # transformation
-    trans = BERTDatasetTransform(
-        tokenizer,
-        max_len,
-        labels=task.get_labels(),
-        pad=pad,
-        pair=task.is_pair,
-        label_dtype='float32' if not task.get_labels() else 'int32')
-
-    data_train = task('train').transform(trans, lazy=False)
+    """Train/eval Data preparation function."""
+    pool = multiprocessing.Pool()
+
+    # transformation for data train and dev
+    label_dtype = 'float32' if not task.class_labels else 'int32'
+    trans = BERTDatasetTransform(tokenizer, max_len,
+                                 class_labels=task.class_labels,
+                                 pad=pad, pair=task.is_pair,
+                                 has_label=True)
+
+    # data train
+    # task.dataset_train returns (segment_name, dataset)
+    train_tsv = task.dataset_train()[1]
+    data_train = mx.gluon.data.SimpleDataset(pool.map(trans, train_tsv))
```

Comment:
I don't recall any problem with python multiprocessing on windows. Could you be more specific? 
I still find mp necessary for task like MNLI, otherwise, it takes minutes to create the dataloader.

Reply:
I did another test and the code was not wrong. The reason for my previous mistake was that there was no code protection.